packages/entity-server/src/routes/hierarchy/middleware/fields.ts

```diff
@@ -40,81 +37,26 @@ export const extractFieldsFromQuery = (queryFields?: string) => {
+      throw new Error(`Unknown field requested: ${requestedField}, must be one of: ${validFields}`);
     }
   });
   return Array.from(fields);
 };
 
-export const mapEntityToFields = (fields: (keyof ExtendedEntityFields)[]) => async (
-  entity: EntityType,
-  context: HierarchyContext,
-) => {
-  const responseBuilder = new EntityResponseObjectBuilder();
-  for (let i = 0; i < fields.length; i++) {
-    const field = fields[i];
-    if (isExtendedField(field)) {
-      responseBuilder.set(field, await extendedFieldFunctions[field](entity, context));
-    } else {
-      responseBuilder.set(field, entity[field]);
-    }
+const flattableFields: (keyof FlattableEntityFields)[] = ['id', 'code', 'name'];
+const isFlattableField = (field: string): field is keyof FlattableEntityFields =>
+  (flattableFields as string[]).includes(field);
+
+export const extractFlatFromQuery = (queryField?: string) => {
+  if (!queryField) {
+    return undefined;
   }
-  return responseBuilder.build();
-};
 
-export const mapEntitiesToFields = (
-  req: HierarchyRequest,
-  fields: (keyof ExtendedEntityFields)[],
-) => async (entities: EntityType[], context: HierarchyContext) => {
-  const relationRecords =
-    fields.includes('parent_code') || fields.includes('child_codes')
-      ? await req.models.ancestorDescendantRelation.getImmediateRelations(context.hierarchyId, {
-          'descendant.country_code': req.ctx.allowedCountries,
-        })
-      : [];
-  const responseBuilders: EntityResponseObjectBuilder[] = new Array(entities.length)
-    .fill(0)
-    .map(() => new EntityResponseObjectBuilder()); // fill array with empty objects
-  for (let i = 0; i < fields.length; i++) {
-    const field = fields[i];
-    if (isExtendedField(field)) {
-      switch (field) {
-        case 'parent_code': {
-          const childToParentMap = reduceToDictionary(
-            relationRecords,
-            'descendant_code',
-            'ancestor_code',
-          );
-          entities.forEach((entity, index) =>
-            responseBuilders[index].set('parent_code', childToParentMap[entity.code]),
-          );
-          break;
-        }
-        case 'child_codes': {
-          const parentToChildrenMap = reduceToArrayDictionary(
-            relationRecords,
-            'ancestor_code',
-            'descendant_code',
-          );
-          entities.forEach((entity, index) =>
-            responseBuilders[index].set('child_codes', parentToChildrenMap[entity.code]),
-          );
-          break;
-        }
-        default: {
-          await Promise.all(
-            entities.map(async (entity, index) => {
-              responseBuilders[index].set(
-                field,
-                await extendedFieldFunctions[field](entity, context),
-              );
-            }),
-          );
-        }
-      }
-    } else {
-      entities.forEach((entity, index) => responseBuilders[index].set(field, entity[field]));
-    }
```

Comment:
Didn't change any functionality here really, just moved to `packages/entity-server/src/routes/hierarchy/format/formatEntityForResponse.ts`